classes/sota_sanity.bbclass

```diff
@@ -10,6 +24,40 @@ def sota_check_required_variables(status, d):
         if not d.getVar(var):
             status.addresult("%s should be set in your local.conf.\n" % var)
 
+def sota_check_variables_validity(status, d):
+    if re.match("^[a-zA-Z0-9_-]*$", d.getVar("OSTREE_BRANCHNAME")) is None:
+        status.addresult("OSTREE_BRANCHNAME Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if re.match("^[a-zA-Z0-9_-]*$", d.getVar("SOTA_HARDWARE_ID")) is None:
+        status.addresult("SOTA_HARDWARE_ID Should only contain characters from the character set [a-zA-Z0-9_-].\n")
+    if d.getVar("SOTA_CLIENT_FEATURES") not in ("hsm", "secondary-network", ""):
+        status.addresult("SOTA_CLIENT_FEATURES should be set to hsm or secondary-network.\n")
+    if d.getVar("SOTA_CLIENT_PROV") not in ("aktualizr-shared-prov", "aktualizr-device-prov", "aktualizr-device-prov-hsm", ""):
+        status.addresult("Valid options for SOTA_CLIENT_PROV are aktualizr-shared-prov, aktualizr-device-prov, and aktualizr-device-prov-hsm.\n")
+    if re.match("^(https?|ftp|file)://.+$", d.getVar("GARAGE_TARGET_URL")) is None:
+        status.addresult("GARAGE_TARGET_URL is set to a bad url.\n")
+    if re.match("^^((([0-9]{3}[1-9]|[0-9]{2}[1-9][0-9]{1}|[0-9]{1}[1-9][0-9]{2}|[1-9][0-9]{3})-(((0[13578]|1[02])-(0[1-9]|[12][0-9]|3[01]))|((0[469]|11)-(0[1-9]|[12][0-9]|30))|(02-(0[1-9]|[1][0-9]|2[0-8]))))|((([0-9]{2})(0[48]|[2468][048]|[13579][26])|((0[48]|[2468][048]|[3579][26])00))-02-29))T([0-1]?[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])Z$", d.getVar("GARAGE_TARGET_EXPIRES")) is None:
```

Comment:
This bit was stole from internet, but it's ugly.

Reply:
While the intention is good, I think this might be going too far. `garage-sign` should already be verifying if this value is correctly formatted and complaining appropriately. The goal of having these checks up front is to prevent problems that are otherwise hard to detect or not obvious. The perfect example is spaces or "unusual" characters in `OSTREE_BRANCHNAME`, which I think has burned several of us at different times. But these inputs to garage-sign have a much more limited use and AFAIK are already verified appropriately.

Reply:
Ok, I removed those two garage-sign checks.